Script to generate failures histogram

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118008

Generates something that looks like
https://gist.github.com/zou3519/43aa8ef28a327bd68cfbac83d84c0999

zou3519 committed:
Script to generate failures histogram

Generates something that looks like
https://gist.github.com/zou3519/43aa8ef28a327bd68cfbac83d84c0999

[ghstack-poisoned]

oulgen commented on scripts/compile_tests/failures_histogram.py:
```diff
@@ -0,0 +1,94 @@
+    raise AssertionError("no message?")
+
+
+IGNORED_REASONS = {
```
Should we mention why we skip some of these? Like e.g. why is "Skipped" or "Skipped!" skipped?

oulgen commented on scripts/compile_tests/failures_histogram.py:
```diff
@@ -0,0 +1,94 @@
+    "Skipped",
+    "Skipped!",
+    "",
+    "test doesn't currently work with dynamo",
```
this seems like a good reason to fix?

zou3519 replied:
I should probably just delete most things from this list.

oulgen commented on scripts/compile_tests/failures_histogram.py:
```diff
@@ -0,0 +1,94 @@
+
+
+def repro(testcase):
+    return f"PYTORCH_TEST_WITH_DYNAMO=1 pytest {testcase.attrib['file']} -v -k {testcase.attrib['name']}"
```
for future, might be good to give file path relative to project root

zou3519 committed:
Update on "Script to generate failures histogram"


Generates something that looks like
https://gist.github.com/zou3519/43aa8ef28a327bd68cfbac83d84c0999

[ghstack-poisoned]

zou3519 committed:
Update on "Script to generate failures histogram"


Generates something that looks like
https://gist.github.com/zou3519/43aa8ef28a327bd68cfbac83d84c0999

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge -f "not tested in CI"
